include/ignition/math/MassMatrix3.hh

```diff
@@ -1106,6 +1106,107 @@ namespace ignition
+                                      _length, _radius, _rot);
+      }
+
+      /// \brief Set inertial properties based on mass and equivalent capsule
+      /// aligned with Z axis.
+      /// \param[in] _mass Mass to set.
+      /// \param[in] _length Length of capsule along Z axis.
+      /// \param[in] _radius Radius of capsule.
+      /// \param[in] _rot Rotational offset of equivalent capsule.
+      /// \return True if inertial properties were set successfully.
+      public: bool SetFromCapsuleZ(const T _mass,
+                                   const T _length,
+                                   const T _radius,
+                            const Quaternion<T> &_rot = Quaternion<T>::Identity)
+      {
+        // Check that _mass, _radius and _length are strictly positive
+        // and that quaternion is valid
+        if (_mass <= 0 || _length <= 0 || _radius <= 0 ||
+            _rot == Quaternion<T>::Zero)
+        {
+          return false;
+        }
+        this->SetMass(_mass);
+        return this->SetFromCapsuleZ(_length, _radius, _rot);
+      }
+
+      /// \brief Set inertial properties based on equivalent capsule
+      /// aligned with Z axis using the current mass value.
+      /// \param[in] _length Length of capsule along Z axis.
+      /// \param[in] _radius Radius of capsule.
+      /// \param[in] _rot Rotational offset of equivalent capsule.
+      /// \return True if inertial properties were set successfully.
+      public: bool SetFromCapsuleZ(const T _length,
+                                   const T _radius,
+                                   const Quaternion<T> &_rot)
+      {
+        // Check that _mass and _size are strictly positive
+        // and that quaternion is valid
+        if (this->Mass() <= 0 || _length <= 0 || _radius <= 0 ||
+            _rot == Quaternion<T>::Zero)
+        {
+          return false;
+        }
+
+        // Mass of cylinder and hemispheres
+        T sphereMass = this->Mass() / (1. + 0.75 * _length / _radius);
+        T cylinderMass = this->Mass() - sphereMass;
+
+        // Diagonal matrix L with principal moments
+        T radius2 = std::pow(_radius, 2);
+        Matrix3<T> L;
+
+        // Inertia components of cylinder
+        L(0, 0) = cylinderMass / 12.0 * (3*radius2 + std::pow(_length, 2));
+        // the following line is repeated later in this function
+        // L(1, 1) = L(0, 0);
+        L(2, 2) = cylinderMass / 2.0 * radius2;
+
+        // Inertia components of hemispheres
+        //  Centroid of hemisphere:
+        //    3/8 radius from flat base on axis of symmetry
+        //  Moment of inertia about centroid:
+        //    2/5 mass * radius^2 about axis of symmetry
+        //    83/320 mass * radius^2 about other axis
+        //    efunda.com/math/solids/solids_display.cfm?SolidName=Hemisphere
+        // add inertia of hemispheres about their centroid
+        L(0, 0) += sphereMass * 83. / 320. * radius2;
```

Comment:
While these magic numbers are traceable, they seem a bit too magical, when really it's just the parallel-axis theorem with some solids (cylinder + hemisphere)?

Rather than bake this all together (which yeah, sure, it's great for efficiency, but how often will this be called?), why not leverage ign math to do parallel-axis with the constituent inertias?
Then you don't need comments, b/c the code is clear.

Reply:
I cleaned this up a bit in 0ac1488